README.md

```diff
@@ -1,4 +1,208 @@
 eos-knowledge-lib
 =================
 
-GUI library for knowledge apps
+Welcome to **eos-knowledge-lib**!
+This is Endless' Javascript library for building offline content apps using modular UI blocks and declarative UI files.
+
+This project uses the GNOME technology stack, including GTK and WebKit for UI, and GJS for its Javascript interpreter.
+
+Building
+--------
+We recommend using [JHbuild](https://developer.gnome.org/jhbuild/stable/) to build this code, especially if you are planning to make changes to it.
+A sample JHbuild moduleset is included in the root directory of this repository.
+
+First download the moduleset with:
+```sh
+wget https://raw.githubusercontent.com/endlessm/eos-knowledge-lib/master/eos-knowledge-lib.modules
+```
+
+Then edit `~/.jhbuildrc` to get JHbuild to use the provided moduleset.
+Here's an example jhbuildrc file:
+```python
+# Change this path to the directory where you put the moduleset file
+modulesets_dir = os.path.expanduser('~')
+moduleset = ['eos-knowledge-lib']
+modules = ['eos-knowledge-lib']
+
+# Change these two paths to your liking
+checkoutroot = os.path.expanduser('~/checkout')
+prefix = os.path.expanduser('~/install')
+
+# Set options to your liking
+os.environ['CFLAGS'] = '-g -Og -fdiagnostics-color=auto'
+makeargs = '-j8'
+use_local_modulesets = True
+```
+
+To build, then do:
+```sh
+jhbuild sysdeps --install
+jhbuild build
+```
+
+When you go into a `jhbuild shell` you will also have to make sure `xapian-bridge` is running in the background:
+```sh
+xapian-bridge &
+```
+
+To run the tests:
+```sh
+jhbuild make check
+```
+
+Building without JHbuild
+------------------------
+
+Alternatively, if you don't want to use JHbuild, you first need to check out and build four other Endless repositories:
+
+ - [eos-sdk](https://github.com/endlessm/eos-sdk)
+ - [eos-shard](https://github.com/endlessm/eos-shard)
+ - [xapian-glib](https://github.com/endlessm/xapian-glib)
+ - [xapian-bridge](https://github.com/endlessm/xapian-bridge)
+
+In addition, to run the tests, you will need to check out and build [jasmine-gjs](https://github.com/ptomato/jasmine-gjs).
+Other suggested dependencies are PyYAML for Python 3, and an SCSS compiler.
+Install these using your system's package manager.
+
+Once you have all the dependencies installed, here's how to build:
```

Comment:
You also have to ensure that `/usr/local` is included in the path for basically everything:

 * PKG_CONFIG_PATH (/usr/local/share/pkgconfig:/usr/local/lib/pkgconfig)
 * LD_LIBRARY_PATH (/usr/local/lib)
 * PATH (/usr/local/bin)

Honestly, I would not recommend to avoid JHbuild; if you can build without JHbuild chances are that you already know how to build. I'm pretty sure that people will follow the second part ("I don't know what JHbuild is, so I don't need it") and then hose their systems and come asking for help.